lib/matplotlib/mlab.py

```diff
@@ -3874,23 +3874,17 @@ def contiguous_regions(mask):
     """
     return a list of (ind0, ind1) such that mask[ind0:ind1].all() is
     True and we cover all such regions
-
-    TODO: this is a pure python implementation which probably has a much
-    faster numpy impl
     """
+    mask = np.asarray(mask, dtype=bool)
+
+    if not mask.size:
+        return []
 
-    in_region = None
-    boundaries = []
-    for i, val in enumerate(mask):
-        if in_region is None and val:
-            in_region = i
-        elif in_region is not None and not val:
-            boundaries.append((in_region, i))
-            in_region = None
+    idx, = np.nonzero(mask[:-1] != mask[1:])
+    idx = np.concatenate((([0],) if mask[0] else ()) + (idx + 1,) +
```

Comment:
could you add a comment here to the effect of "prepending index 0 and appending index len(mask) if needed" (at least, that is my understanding of what it is doing. Too many parens.

In fact, perhaps you could have them be lists instead of tuples to help distinguish when I am looking at a sequence and when we are merely grouping for order of operations?